src/main/webapp/app/entities/programming-exercise/services/programming-exercise-test-case.service.ts

```diff
@@ -47,12 +50,44 @@ export class ProgrammingExerciseTestCaseService implements IProgrammingExerciseT
         }
     }
 
+    /**
+     * Send new values for the test cases of an exercise to all subscribers.
+     * @param exerciseId
+     * @param testCases
+     */
+    public notifyTestCases(exerciseId: number, testCases: ProgrammingExerciseTestCase[]): void {
+        if (this.subjects[exerciseId]) {
+            this.subjects[exerciseId].next(testCases);
+        }
+    }
+
     /**
      * Executes a REST request to the test case endpoint.
      * @param exerciseId
      */
     private getTestCases(exerciseId: number): Observable<ProgrammingExerciseTestCase[]> {
-        return this.http.get<ProgrammingExerciseTestCase[]>(`${this.testCaseUrl}/${exerciseId}`);
+        return this.http.get<ProgrammingExerciseTestCase[]>(`${this.testCaseUrl}/${exerciseId}/test-cases`);
+    }
+
+    /**
+     * Update the weight with the provided value of the provieded test case.
+     * Needs the exercise to verify permissions on the server.
+     *
+     * @param exerciseId
+     * @param testCaseId
+     * @param weight
+     */
+    public updateWeight(exerciseId: number, testCaseId: number, weight: number): Observable<ProgrammingExerciseTestCase> {
+        return this.http.patch<ProgrammingExerciseTestCase>(`${this.testCaseUrl}/${exerciseId}/test-cases/${testCaseId}/update-weight`, { weight });
```

Comment:
would it not be more efficient to update all test case weights at the same time? Instead of updating each single entry? So could you provide a method updateWeights?

Reply:
It would be more efficient, but it would make both the client and the server code more complicated:
Both would then have to use a dto to communicate which test cases could be updated and which could not be, even though it is very unlikely that this patch call fails if executed from the Angular client and not manually from a REST client.

When I implemented this, I figured that only instructors will use this feature, so I made the implementation light weight.

Reply:
After looking into the server side a bit more, I decided to change the behavior to saving all test cases at once.

Reply:
Implementation done.